lib/router/mbaas-auth.js

```diff
@@ -0,0 +1,39 @@
+const q = require('q');
+
+/**
+ * Auth sends a request on 'wfm:user:auth' topic, raincatcher-demo-auth is subscriber
+ * of this topic and verifies passed in credentials.
+ *
+ * @param {Object} mediator object used to subscribe and publish read and auth topics
+ * @param {String} username Unique identifier of the user.
+ * @param {String} password passed in password to be verified against user.
+ *
+ */
+exports.auth = function(mediator, username, password) {
+
+  /**
+   * Returns promise with user profile for authenticated user, or an error
+   * if user doesn't exist or is not authenticated.
+   *
+   * @param {boolean} authenticated
+   * @returns {*|promise} containing user profile for authenticated user.
+   */
+  function checkAuthentication(authenticated) {
```

Comment:
@witmicko Might want to move this out of the function. It would read much better and be easier to test.